test/core/external/lib/mockZeroExOrderDataHandlerLibrary.spec.ts

```diff
@@ -0,0 +1,244 @@
+    feeRecipientAddress,
+    senderAddress,
+    makerAssetAmount,
+    takerAssetAmount,
+    makerFee,
+    takerFee,
+    expirationTimeSeconds,
+    salt,
+    makerAssetData,
+    takerAssetData,
+  );
+
+  // Header Subject Data
+  let signatureLength: UInt = new BigNumber(signature.length);
+  let zeroExOrderLength = new BigNumber(0);
+  let makerAssetDataLength = new BigNumber(makerAssetData.length);
+  let takerAssetDataLength = new BigNumber(takerAssetData.length);
+
+  beforeEach(async () => {
+    const zeroExExchangeWrapperInstance = await MockZeroExOrderDataHandlerLibrary.new(
+      { from: ownerAccount, gas: DEFAULT_GAS },
+    );
+
+    zeroExExchangeWrapper = new MockZeroExOrderDataHandlerLibraryContract(
+      web3.eth.contract(zeroExExchangeWrapperInstance.abi).at(zeroExExchangeWrapperInstance.address),
+      { from: ownerAccount },
+    );
+  });
+
+  describe("#getOrderDataHeader", async () => {
+    let subjectOrderData: Bytes32;
+
+    beforeEach(async () => {
+      subjectOrderData = bufferArrayToHex(bufferOrderHeader(
+        signatureLength,
+        zeroExOrderLength,
+        makerAssetDataLength,
+        takerAssetDataLength
+      ));
+    });
+
+    it("works", async () => {
+      const result = await zeroExExchangeWrapper.getOrderDataHeader.callAsync(subjectOrderData);
+
+      expect(result[0]).to.bignumber.equal(signatureLength);
+      expect(result[1]).to.bignumber.equal(zeroExOrderLength);
+      expect(result[2]).to.bignumber.equal(makerAssetDataLength);
+      expect(result[3]).to.bignumber.equal(takerAssetDataLength);
+    });
+  });
+
+  describe("#getFillAmount", async () => {
+    let subjectOrderData: Bytes32;
+
+    beforeEach(async () => {
+      subjectOrderData = bufferArrayToHex(
+        bufferOrderHeader(
+          signatureLength,
+          zeroExOrderLength,
+          makerAssetDataLength,
+          takerAssetDataLength,
+        ).concat(bufferFillAmount(fillAmount)
+      ));
+    });
+
+    it("works", async () => {
+      const result = await zeroExExchangeWrapper.getFillAmount.callAsync(subjectOrderData);
+      expect(result).to.be.bignumber.equal(fillAmount);
+    });
+  });
+
+  describe("#getSignature", async () => {
+    let subjectOrderData: Bytes32;
+
+    beforeEach(async () => {
+      subjectOrderData = bufferArrayToHex(
+        bufferOrderHeader(
+          signatureLength,
+          zeroExOrderLength,
+          makerAssetDataLength,
+          takerAssetDataLength,
+        )
+        .concat(bufferFillAmount(fillAmount))
+        .concat(bufferSignature(signature)
+      ));
+    });
+
+    it("works", async () => {
+      const result = await zeroExExchangeWrapper.getSignature.callAsync(subjectOrderData);
+      expect(web3.toAscii(result)).to.equal(signature);
+    });
+  });
+
+  describe("#parseZeroExOrder", async () => {
+    let subjectOrderData: Bytes32;
+
+    beforeEach(async () => {
+      const zeroExOrderBuffer = bufferZeroExOrder(zeroExOrder);
```

Comment:
move this setup into `test/utils/zeroExExchangeWrapper.ts`. give it a descriptive name